src/detect-engine-tag.c

```diff
@@ -901,35 +878,26 @@ static int DetectTagTestPacket03 (void)
             expect = false;
         else
             expect = true;
-        if (((p[i]->flags & PKT_HAS_TAG) ? true : false) != expect)
-            goto cleanup;
+        FAIL_IF(((p[i]->flags & PKT_HAS_TAG) ? true : false) != expect);
```

Comment:
Would this look better as
 `FAIL_IF_NOT(((p[i]->flags & PKT_HAS_TAG) ? true : false) == expect);`?

Reply:
There is an `if` block right above this block,
should I convert it too?

One way I could think of converting it would be
to do something like this;
```
    /* see if the PKT_HAS_TAG is set on the packet if needed */
    FAIL_IF(((p[0]->flags & PKT_HAS_TAG) ? true : false) != false);
    FAIL_IF(((p[1]->flags & PKT_HAS_TAG) ? true : false) != true);
    FAIL_IF(((p[2]->flags & PKT_HAS_TAG) ? true : false) != false);
    FAIL_IF(((p[3]->flags & PKT_HAS_TAG) ? true : false) != false);
    FAIL_IF(((p[4]->flags & PKT_HAS_TAG) ? true : false) != true);
    FAIL_IF(((p[5]->flags & PKT_HAS_TAG) ? true : false) != false);
```
but that would be outside the `for` block, and I think that defeats the
purpose of the block, right?

Reply:
lets keep the loop, but remove the branching:
`bool expect = (i == 0 || i == 3 || i == 5 || i == 6);`